src/libcore/atomic.rs

```diff
@@ -1089,3 +1090,23 @@ pub fn fence(order: Ordering) {
         }
     }
 }
+
+macro_rules! impl_Debug {
+    ($($t:ident)*) => ($(
+        #[stable(feature = "atomic_debug", since = "1.3.0")]
+        impl fmt::Debug for $t {
+            fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+                f.debug_tuple(stringify!($t)).field(&self.load(Ordering::Relaxed)).finish()
```

Comment:
Not necessarily, since the core running this debug code (for example, perhaps a monitor thread) and the code doing computation that it is synchronizing with the atomic are separate.